### PR TITLE
Add UV_REQUEST_TIMEOUT argument/envvar to building CI/PROD images

### PR DESCRIPTION
**Pull request by potiuk**
UV has smaller timeout than PIP and in some cases it might be too low. This PR adds flag with ENVVAR that allows to set custom value for the envvar while building the image.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,6 +52,8 @@ ARG PYTHON_BASE_IMAGE="python:3.8-slim-bookworm"
 ARG AIRFLOW_PIP_VERSION=24.0
 ARG AIRFLOW_UV_VERSION=0.1.24
 ARG AIRFLOW_USE_UV="false"
+# timeout for UV Requests
```
Comment (uranusjr) [inline Dockerfile]: Not particularly sure about this comment. The variable name seems self-explainatory, and we don’t comment other arguments similarly.

File: dev/breeze/src/airflow_breeze/commands/common_options.py [hunk 2]
```diff
@@ -354,6 +354,14 @@ def _set_default_from_parent(ctx: click.core.Context, option: click.core.Option,
     help="Use uv instead of pip as packaging tool to build the image.",
     envvar="USE_UV",
 )
+option_uv_request_timeout = click.option(
+    "--uv-request-timeout",
+    help="Timeout for requests that UV makes (only used in case of UV builds).",
+    type=click.IntRange(1, 10000),
```
Comment (uranusjr) [inline dev/breeze/src/airflow_breeze/commands/common_options.py]: Why the specific 10000 max…?

**Review by uranusjr — APPROVED** [review]
A couple of nits. lgtm.

File: Dockerfile (hunk 1 above)
Comment (jedcunningham) [reply Dockerfile]: Yep, I think we should axe it.

**Review by jedcunningham — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/commands/common_options.py (hunk 2 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/common_options.py]: No particular reasons but ~ 3 hours should be rather max :) . Removed max.

File: Dockerfile (hunk 1 above)
Comment (potiuk) [reply Dockerfile]: Yeah . It's like D105 :). Paints the object.

**Review by jedcunningham — APPROVED** [review]

**Pull request merged**